metricflow/plan_conversion/to_sql_plan/dataflow_to_subquery.py

```diff
@@ -215,9 +216,19 @@ def _make_time_spine_data_set(
         ]
         required_specs = queried_specs + specs_required_for_where_constraints
 
-        time_spine_source = TimeSpineSource.choose_time_spine_source(
+        time_spine_sources = TimeSpineSource.choose_time_spine_sources(
             required_time_spine_specs=required_specs, time_spine_sources=self._time_spine_sources
         )
+        if len(time_spine_sources) != 1:
+            raise RuntimeError(
+                str(
+                    LazyFormat(
+                        "Unexpected number of time spine sources required for query - cumulative metrics should require exactly one.",
```

Comment:
Custom grains are joined later for cumulative metrics, so this should always be true.